### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by yns88**
I've been meaning to do this ever since we got access to Actions; now that Travis is acting up again I figured I might as well.

### CODE REVIEW

File: .github/workflows/setup.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -ex
+
+add-apt-repository ppa:git-core/ppa
```
Comment (bigfootjon) [inline .github/workflows/setup.sh:4]: 4.32-latest fails in a way that the DataProvider fix in #93 won't fix. I think we still need a newer version of hg:
```bash
UBUNTU_VERSION=$(lsb_release -r -s)
if [ $(echo $UBUNTU_VERSION | cut -c1-2) -lt "18" ]; then
  add-apt-repository ppa:mercurial-ppa/releases
fi
```
(from `.travis.sh`)

Comment (yns88) [reply .github/workflows/setup.sh:4]: Dang, I guess the HHVM docker images must be running an old version of Ubuntu still.

Comment (yns88) [reply .github/workflows/setup.sh:4]: Yikes, 14.04 https://hub.docker.com/r/hhvm/hhvm

**Comment by bigfootjon** [comment]
Wait that paragraph on the Hub is out of date, the [Dockerfile](https://github.com/hhvm/hhvm-docker/blob/master/hhvm-latest/Dockerfile) for HHVM is using 18.04

Something else must be wrong with hg

**Comment by yns88** [comment]
Right, according to logs the versions of hg and all HHVM packages are all the same as the most recent successful job on Travis, so it's unclear to me what could be causing that failure...

**Pull request closed without merging**